symengine/parser/parser.cpp

```diff
@@ -4,10 +4,29 @@
 #include <symengine/real_mpfr.h>
 #include <symengine/ntheory_funcs.h>
 #include <symengine/parser/tokenizer.h>
+#if ((defined(_MSVC_LANG) && _MSVC_LANG >= 201703L) || __cplusplus >= 201703L)
+#include <charconv>
+#endif
 
 namespace SymEngine
 {
 
+static double string_to_double(const char *str, char **endptr)
+{
+#if ((defined(_MSVC_LANG) && _MSVC_LANG >= 201703L) || __cplusplus >= 201703L)
+    // use std::from_chars: locale-independent, non-allocating, non-throwing
+    double result{0.0};
```

Comment:
Do we expect this function to ever fail? If not do we need to initialize `result` here?

Reply:
Ok, I was just thinking that if the initialization doesn't matter because the parser guarantees that this is indeed a parsable string it could be removed to speed up the function a tiny bit more.